### PR TITLE
Upgrade to CAPA v2.5.2

### PR DESCRIPTION
**Pull request by fiunchinho**
### Checklist

- [X] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by fiunchinho** [comment]
`gitleaks` fails due to some example dummy AWS ACCESS KEY, it can be ignored.

File: helm/cluster-api-provider-aws/values.yaml [hunk 1]
```diff
@@ -4,23 +4,14 @@ name: cluster-api-provider-aws
...
-tag: v2.4.2-gs-70b4664c1
+# Changes in this tag include upstream v2.5.2 with additional:
+#   * GiantSwarm fork modifications (https://github.com/giantswarm/cluster-api-provider-aws/pull/607/commits/08280ec433b63119d652b39c06d9a4b78d5a9055)
+#   * when aws-cni is disabled deleted all resources including helm managed resources (https://github.com/giantswarm/cluster-api-provider-aws/pull/607/commits/bbbb36a7a7bd050bb232bfa60a10b4202ac99850)
+#   * do not place EKS nodes in the CNI subnets (https://github.com/giantswarm/cluster-api-provider-aws/pull/607/commits/edd14fb6bd39434a249ed8696e0abeb9b85f6c35)
+#   * add non root volumes to AWSMachinePools (https://github.com/giantswarm/cluster-api-provider-aws/pull/607/commits/ab0528f4b80ed36fd7b508c4137ee09d41fefa8c)
+#   * support adding custom secondary VPC CIDR blocks in `AWSCluster` (https://github.com/giantswarm/cluster-api-provider-aws/pull/607/commits/ae00674b1ed47e0e939f44d4712507839df97106)
+#   * S3 user data support for AWSMachinePools (https://github.com/giantswarm/cluster-api-provider-aws/pull/607/commits/905c44ab2afde5e2d4e983066ab9856859a0de47)
+tag: v2.5.2-gs-905c44ab2
```
Comment (AndiDog) [inline helm/cluster-api-provider-aws/values.yaml:14]: We should first merge https://github.com/giantswarm/cluster-api-provider-aws/pull/607 and then tag the merge commit. I'm looking into it.

LGTM until here.

Comment (fiunchinho) [reply helm/cluster-api-provider-aws/values.yaml:14]: > We should first merge [giantswarm/cluster-api-provider-aws#607](https://github.com/giantswarm/cluster-api-provider-aws/pull/607) and then tag the merge commit. I'm looking into it.

Definitely.

Comment (fiunchinho) [reply helm/cluster-api-provider-aws/values.yaml]: Done

**Review by AndiDog — APPROVED** [review]

**Pull request merged**